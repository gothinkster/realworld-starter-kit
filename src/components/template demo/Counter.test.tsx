import { describe, expect, test } from "vitest";
import { render, fireEvent } from "solid-testing-library";
import Counter from "./Counter";

describe("<Counter />", () => {
  test("renders", () => {
    const { container, unmount } = render(() => <Counter />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  test("increments value", async () => {
    const { queryByRole, unmount } = render(() => <Counter />);
    const button = (await queryByRole("button")) as HTMLButtonElement;
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(/Clicks: 0/);
    fireEvent.click(button);
    expect(button).toHaveTextContent(/Clicks: 1/);
  });
});
