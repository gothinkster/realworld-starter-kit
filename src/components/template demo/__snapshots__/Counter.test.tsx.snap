// Vitest Snapshot v1

exports[`<Counter /> > renders 1`] = `
<div>
  <button
    class="increment"
  >
    Clicks: 
    <!--#-->
    0
    <!--/-->
  </button>
</div>
`;
